v2/internal/typescriptify/typescriptify.go

```diff
@@ -836,7 +836,7 @@ func (t *typeScriptClassBuilder) AddStructField(fieldName string, field reflect.
 	namespace := strings.Split(field.Type.String(), ".")[0]
 	fqname := "any"
 	classname := "null"
-	fqname = field.Type.Name()
+	fqname = t.prefix + field.Type.Name() + t.suffix
 	if namespace != t.namespace {
 		fqname = field.Type.String()
```

Comment:
I think that we are forgetting this scenario as well 🙂 
```suggestion
		fqname = t.prefix + field.Type.String() + t.suffix
```

Reply:
The best way to test this is to add more scenarios to the existing generation tests 👍

Reply:
What do you think about this scenario that uses nested namespaces? 

## App scenario

**Package 1**
```go
package myPackage1

type MyStruct struct {
  id  string
  ref MyOtherStruct
}

type MyOtherStruct struct{
   id string
}
```

**Package 2**
```go
package myPackage2

type MyStruct struct {
  id  string
  ref myPackage1.MyStruct 
  refArray []myPackage1.MyStruct 
}
```

**App to bind**
```go
type App struct {}

func (a *App) Greet(arg myPackage2.MyStruct) myPackage2.MyStruct {
  return myPackage2.MyStruct{}
}
```

## Expected generation

**Prefix:** ThisIs
**Suffix:** Model

**models.ts**
```typescript
namespace myPackage1 {
   class ThisIsMyStructModel {
      id: string;
      ref: ThisIsMyOtherStructModel;
   }

   class ThisIsMyOtherStructModel {
      id: string;
   }
}

namespace myPackage2 {
   class ThisIsMyStructModel {
      id: string;
      ref: myPackage1.ThisIsMyStructModel;
      refArray: myPackage1.ThisIsMyStructModel[];
   }
}
```

**App.d.ts**
```typescript
import ...

export function Greet(arg1:myPackage2.ThisIsMyStructModel):Promise<myPackage2.ThisIsMyStructModel>;
```

Reply:
Thank you very much @Goncalo-Marques , that's helpful!
This was not correctly handled, so fixing this in the PR

arguments in the function signature were also not correctly handled, i.e. `export function Greet(arg1:myPackage2.MyStruct)` where type of `arg1` was not correctly handled fixed as well

Added some tests

Reply:
I don't have a lot of insight into this as well and learning about it as I dive into this, therefore ideas are highly appreciated :)